### PR TITLE
DOMA-1707 push sbbol events to sales crm + tests

### PR DESCRIPTION
**Pull request by MrFoxPro**

### CODE REVIEW

File: apps/condo/domains/organization/integrations/sbbol/sync/syncOrganization.js [hunk 1]
```diff
@@ -105,6 +105,7 @@ const syncOrganization = async ({ context, user, userInfo, organizationInfo, dvS
         const existingOrganization = userOrganizations.find(({ meta }) => meta.inn === organizationInfo.meta.inn)
         if (!existingOrganization) {
             const newOrganization = await createOrganization({ context, user, organizationInfo, dvSenderFields })
+            pushOrganizationToSalesCRM(newOrganization, true)
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncOrganization.js:108]: We already sent the event inside createOrganisation.

Comment (pahaz) [reply apps/condo/domains/organization/integrations/sbbol/sync/syncOrganization.js:108]: It’s looks like redundant action

File: apps/condo/domains/organization/utils/serverSchema/Organization.js [hunk 2]
```diff
@@ -93,19 +94,43 @@ async function findOrganizationEmployee (context, query) {
 
 const salesCRMRequestLogger = pino({ name: 'sales_crm', enabled: falsey(process.env.DISABLE_LOGGING) })
 
-async function pushToSalesCRM (organization) {
-    if (!SALES_CRM_WEBHOOK_URL) {
-        return salesCRMRequestLogger.warn('SALES_CRM_WEBHOOK_URL not specified correctly in config')
+async function pushOrganizationToSalesCRM (organization, fromSbbol = false) {
```
Comment (pahaz) [inline apps/condo/domains/organization/utils/serverSchema/Organization.js:97]: The second argument looks like a hack. Try to use data from the organization

File: apps/condo/domains/organization/integrations/sbbol/sync/syncOrganization.js (hunk 1 above)
Comment (pahaz) [reply apps/condo/domains/organization/integrations/sbbol/sync/syncOrganization.js:108]: It’s better to add something inside the organization context

File: apps/condo/domains/organization/utils/serverSchema/Organization.js [hunk 3]
```diff
@@ -93,19 +94,43 @@ async function findOrganizationEmployee (context, query) {
...
+    if (!SALES_CRM_ORGANIZATIONS_WEBHOOK_URL) {
+        return salesCRMRequestLogger.warn('SALES_CRM_ORGANIZATIONS_WEBHOOK_URL not specified correctly in config')
+    }
+    if (config.NODE_ENV !== 'production') {
```
Comment (pahaz) [inline apps/condo/domains/organization/utils/serverSchema/Organization.js:101]: It’s redundant. If you have this setting you need to send the data.

File: apps/condo/domains/organization/integrations/sbbol/sync/syncSbbolSubscriptionPaymentRequestsState.js [hunk 4]
```diff
@@ -36,14 +37,17 @@ const syncSbbolSubscriptionPaymentRequestStateFor = async (payment, fintechApi)
...
                         ...dvSenderFields,
                         type: SUBSCRIPTION_TYPE.SBBOL,
                         isTrial: false,
                         organization: { connect: { id: payment.organization.id } },
-                        startAt: now.toISOString(),
-                        finishAt: now.add(1, 'year').toISOString(),
+                        startAt,
+                        finishAt,
                     })
+                    pushSubscriptionActivationToSalesCRM(data.payerInn, startAt, finishAt, false)
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncSbbolSubscriptionPaymentRequestsState.js:50]: Why here and not inside afterChange hook ?

Comment (AntonAL) [reply apps/condo/domains/organization/integrations/sbbol/sync/syncSbbolSubscriptionPaymentRequestsState.js:50]: I think, we can use hooks.

File: apps/condo/domains/organization/schema/Organization.js [hunk 5]
```diff
@@ -126,6 +128,18 @@ const Organization = new GQLListSchema('Organization', {
...
         },
     },
+    hooks: {
+        afterChange: ({
+            operation,
+            existingItem,
+            updatedItem,
+        }) => {
+            // check update operation, because importRemoteSystem field sets separately with update operation
+            if (operation === 'update' && existingItem.importRemoteSystem !== SBBOL_IMPORT_NAME && updatedItem.importRemoteSystem === SBBOL_IMPORT_NAME) {
```
Comment (pahaz) [inline apps/condo/domains/organization/schema/Organization.js:138]: What about create operation ?

**Review by AntonAL — APPROVED** [review]

File: apps/condo/domains/organization/integrations/salesCrm.spec.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
+const mockPushOrganizationToSalesCRM = jest.fn()
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/salesCrm.spec.js:1]: move it inside syncOrganiaztion spec.

File: apps/condo/domains/organization/utils/serverSchema/Organization.js [hunk 7]
```diff
@@ -93,31 +96,52 @@ async function findOrganizationEmployee (context, query) {
 
 const salesCRMRequestLogger = pino({ name: 'sales_crm', enabled: falsey(process.env.DISABLE_LOGGING) })
 
-async function pushToSalesCRM (organization) {
-    if (!SALES_CRM_WEBHOOK_URL) {
-        return salesCRMRequestLogger.warn('SALES_CRM_WEBHOOK_URL not specified correctly in config')
+async function pushOrganizationToSalesCRM (organization) {
+    if (!SALES_CRM_ORGANIZATIONS_WEBHOOK_URL) {
+        return salesCRMRequestLogger.warn('SALES_CRM_ORGANIZATIONS_WEBHOOK_URL not specified correctly in config')
```
Comment (pahaz) [inline apps/condo/domains/organization/utils/serverSchema/Organization.js:101]: You don’t need to log it. It’s annoying in development

File: apps/condo/domains/organization/utils/serverSchema/Organization.js [hunk 8]
```diff
@@ -93,31 +96,52 @@ async function findOrganizationEmployee (context, query) {
...
+    try {
+        await axios.post(SALES_CRM_SUBSCRIPTIONS_WEBHOOK_URL, {
+            payerInn,
+            startAt,
+            finishAt,
+            isTrial,
+        })
+    }
+    catch (e) {
+        salesCRMRequestLogger.warn('Request to sales crm failed', e)
```
Comment (pahaz) [inline apps/condo/domains/organization/utils/serverSchema/Organization.js:134]: Use json formst

File: apps/condo/domains/organization/utils/serverSchema/Organization.js (hunk 7 above)
Comment (MrFoxPro) [reply apps/condo/domains/organization/utils/serverSchema/Organization.js]: Done

File: apps/condo/domains/organization/utils/serverSchema/Organization.js (hunk 8 above)
Comment (MrFoxPro) [reply apps/condo/domains/organization/utils/serverSchema/Organization.js]: Done!

File: apps/condo/domains/subscription/schema/ServiceSubscription.js [hunk 9]
```diff
@@ -180,6 +181,11 @@ const ServiceSubscription = new GQLListSchema('ServiceSubscription', {
                 return addValidationError(`${OVERLAPPING_ERROR} subscription for current organization overlaps already existing by its time period`)
             }
         },
+        afterChange: ({ operation, updatedItem }) => {
+            if (operation === 'create' && updatedItem.isTrial === false && updatedItem.sbbolOfferAccept){
```
Comment (pahaz) [inline apps/condo/domains/subscription/schema/ServiceSubscription.js:185]: if update?

**Review by pahaz — APPROVED** [review]

**Pull request merged**